projects/netgrif-components-core/src/lib/side-menu/content-components/user-assign/base-user-assign-list/abstract-base-user-assign-list.component.ts

```diff
@@ -0,0 +1,62 @@
+import {Component, Input, OnDestroy, OnInit, ViewChild} from "@angular/core";
+import {RolesObject, UserListItem, UserListService} from "../../../../user/services/user-list.service";
+import {FormControl} from "@angular/forms";
+import {CdkVirtualScrollViewport} from "@angular/cdk/scrolling";
+import {debounceTime} from "rxjs/operators";
+import {UserValue} from "../../../../data-fields/user-field/models/user-value";
+
+@Component({
+    selector: 'ncc-base-user-assign-list',
+    template: ''
+})
+export abstract class AbstractBaseUserAssignListComponent implements OnInit, OnDestroy {
+
+    protected SEARCH_DEBOUNCE_TIME = 600;
+
+    @Input() searchUserControl: FormControl;
+
+    @Input() roles: RolesObject | Array<string>;
+
+    @Input() negativeRoles: RolesObject | Array<string>;
+
+    @ViewChild(CdkVirtualScrollViewport) public viewport: CdkVirtualScrollViewport;
+
+    constructor(protected _userListService: UserListService) {
+
+    }
+
+    ngOnInit() {
+        this.searchUserControl.valueChanges.pipe(debounceTime(this.SEARCH_DEBOUNCE_TIME)).subscribe(searchText => {
+            this._userListService.reload(searchText);
+        });
+        if (this.roles instanceof Array) {
+            this._userListService.rolesQuery = this.roles;
```

Comment:
where is `UserListService` provided?
If it is provided in root, then this implementation does not allowuser assign list to operate concurently with itself (multiple instances/lists), since the search query is stored in the global shared service.
Is this intended/desired?

Reply:
Same as above, I just copied this code to create an abstraction, this code was already present in another class. As far as I know, the `UserListService` is not provided in root, and is provided directly in `UserAssignComponent` class in `@netgrif/components` library.